apps/aecore/src/aec_fork_signalling.erl

```diff
@@ -0,0 +1,350 @@
+%%%============================================================================
+%%% @copyright (C) 2019, Aeternity Anstalt
+%%% @doc
+%%% H1 - height before the the signalling start height;
+%%% HS - signalling start height;
+%%% HE - signalleing end height;
+%%% H2 - fork height;
+%%%
+%%% H1 < HS < HE < H2
+%%%
+%%% The gen_server process (aec_fork_signalling) makes sure that there is a
+%%% fork result at H2 (true, false or pending). In order to compute the fork
+%%% result, it manages asynchronous processes that walk the chain and compute
+%%% how many blocks include a predefined info field between HE - 1 and HS
+%%% (inclusive).
+%%%
+%%% The block at height HE - 1 is the last signalling block, and there must
+%%% be a result computed for it. Once a worker process that computes the
+%%% result is spawned, a 'pending' atom for the given block hash is  stored in
+%%% results map. Once the worker process is done with the computation it will
+%%% send a boolean result and the 'pending' atom is replaced with the boolean
+%%% result.
+%%%
+%%% The blocks with height between HE and FH (inclusive) are treated
+%%% differently. Once such block arrives, its previous key hash is looked up
```

Comment:
I would expect:
- Identify the block hash of the predecessor key block at height HE - 1.
- If such block hash at height HE - 1 does not have an entry (boolean or pending), spawn worker for it.

Reply:
If a block at height between HE and H2 - 1 arrives:
* if there is an entry in the results map for prev_key_hash, no new process is spawned

Reply:
I see the problem as being space vs. time, especially in identifying "predecessor key block at height HE - 1". 🤔

Reply:
This is resolved as not all the pointers between `HE` and `H2 - 1` are stored anymore.